mapper.cpp

```diff
@@ -462,8 +503,25 @@ vector<Alignment> Mapper::align_threaded(Alignment& alignment, int& kmer_count,
         if (!allATGC(k)) continue; // we can't handle Ns in this scheme
         //if (debug) cerr << "kmer " << k << " entropy = " << entropy(k) << endl;
         if (min_kmer_entropy > 0 && entropy(k) < min_kmer_entropy) continue;
-        uint64_t approx_matches = index->approx_size_of_kmer_matches(k);
-        // Report the approximate match count
+        
+        // We fill this in only once if we're using GCSA indexing
+        gcsa::range_type gcsa_range;
+        
+        // Work out the number of *bytes* of matches for this kmer with the appropriate index.
+        uint64_t approx_matches;
+        if(gcsa) {
+            // A little more complicated. We run the search and count the range size
+            gcsa_range = gcsa->find(k);
+            // Measure count and convert to bytes
+            approx_matches = gcsa::Range::length(gcsa_range) * sizeof(gcsa::node_type);
```

Comment:
I'm not sure we should be using the approximate size of matches in kilobytes, as this is a nice precise estimate. Maybe either use approx_matches or exact_matches. I can change this after merge.